Upgrading Ruby on Rails.1.General Advice (#23)
campuscode/rails-guides-pt-BR: pt-BR/upgrading_ruby_on_rails.md

**DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**

> Atualizando Ruby on Rails

Reviewer: ```suggestion
Atualizando o Ruby on Rails
```

Fiz a sugestão mas ainda estou na dúvida o que acha? @erikacamposdesign

Reviewer 2: Acho necessário colocar o artigo, apesar de em inglês não ser. O texto fica mais fluido.

=======================

Este guia fornece os passos a serem seguidos quando você for atualizar suas aplicações

> para uma versão mais nova do Ruby on Rails. Estes passos também estão disponíveis em guias de releases individuais.

Reviewer: ```suggestion
para uma versão mais nova do Ruby on Rails. Estes passos também estão disponíveis em guias de lançamento de versão individuais.
```

To procurando algo para traduzir, pensei em colocar algo assim. O que acha?

Author: @HenriqueMorato neste caso mantive `releases`  por ser um termo bastante usado e conhecido no meio. Prefere que mude para a tradução? Ou então posso marca-lo como estrangeirismo com itálico.

O que acham?

Reviewer: Acho legal sim! Ainda temos que criar um padrão de tradução mas acho ótimo do itálico.

--------------------------------------------------------------------------------

Conselho Geral
--------------

> Antes de tentar atualizar uma aplicação existente, você deve ter certeza que possui uma boa razão para faze-lo. Então tenha em mente alguns fatores: a necessidade de novas funcionalidades, a crescente dificuldade de encontrar suporte para código mais antigo, seu tempo disponível e habilidades, por exemplo

Reviewer: Acho que aqui faltou o ponto final. Estou pensando em algo para o `to name a few.` talvez `para citar alguns` ou `entre outros`? O que vocês acham? @erikacamposdesign @joaorsalmeida @Marcelo-Zanini

Author: `entre outros` acredito que resolve bem sim.

### Cobertura de Testes

> A melhor maneira de garantir que sua aplicação ainda funciona após a atualização é possuir uma boa cobertura de testes antes de começar o processo. Se você não possuir testes automatizados para a maior parte de sua aplicação, será necessário gastar algum tempo realizando testes manuais de todas as partes alteradas. No caso da atualização Ruby on Rails, isso significa cada umas das funcionalidades dentro da aplicação. Faça a si mesmo um favor e tenha certeza de ter uma boa cobertura de teste _antes_ de iniciar uma atualização.

Reviewer: ```suggestion
A melhor maneira de garantir que sua aplicação ainda funciona após a atualização é possuir uma boa cobertura de testes antes de começar o processo. Se você não possuir testes automatizados para a maior parte de sua aplicação, será necessário gastar algum tempo realizando testes manuais de todas as partes alteradas. No caso da atualização de Rails, isso significa cada umas das funcionalidades dentro da aplicação. Faça a si mesmo um favor e tenha certeza de ter uma boa cobertura de teste _antes_ de iniciar uma atualização.
```

Reviewer: Como estamos começando a convencionar itálico estrangeirismos talvez possamos trocar o `antes` para negrito. @erikacamposdesign tem alguma opinião sobre o assunto?

O que acha @Marcelo-Zanini ?

Author: Também achei que ficaria confuso, mas mantive o itálico neste commit para manter o mais próximo do original. Por se tratar de uma chamada de atenção acredito que negrito fica melhor mesmo.

Reviewer 2: Como estamos usando em itálico as palavras que não serão traduzidas, eu usaria o negrito para dar ênfase

> **DO NOT READ THIS FILE ON GITHUB, GUIDES ARE PUBLISHED ON https://guides.rubyonrails.org.**

Reviewer: ```suggestion
**NÃO LEIA ESTE ARQUIVO NO GITHUB, OS GUIAS SÃO PUBLICADOS NO https://guiarails.com.br.**
```

Atualizando Ruby on Rails
=======================

Este guia fornece os passos a serem seguidos quando você for atualizar suas aplicações
para uma versão mais nova do Ruby on Rails. Estes passos também estão disponíveis em guias de lançamento de versão individuais.

--------------------------------------------------------------------------------

Conselho Geral
--------------

> Antes de tentar atualizar uma aplicação existente, você deve ter certeza que possui uma boa razão para faze-lo. Então tenha em mente alguns fatores: a necessidade de novas funcionalidades, a crescente dificuldade de encontrar suporte para código mais antigo, seu tempo disponível e habilidades, entre outros.

Reviewer: ```suggestion
Antes de tentar atualizar uma aplicação existente, você deve ter certeza que possui uma boa razão para fazê-lo. Então tenha em mente alguns fatores: a necessidade de novas funcionalidades, a crescente dificuldade de encontrar suporte para código mais antigo, seu tempo disponível e habilidades, entre outros.
```

### Cobertura de Testes

> A melhor maneira de garantir que sua aplicação ainda funciona após a atualização é possuir uma boa cobertura de testes antes de começar o processo. Se você não possuir testes automatizados para a maior parte de sua aplicação, será necessário gastar algum tempo realizando testes manuais de todas as partes alteradas. No caso da atualização do Rails, isso significa cada umas das funcionalidades dentro da aplicação. Faça a si mesmo um favor e tenha certeza de ter uma boa cobertura de teste **antes** de iniciar uma atualização.

Reviewer 2: ```suggestion
A melhor maneira de garantir que sua aplicação ainda funciona após a atualização é possuir uma boa cobertura de testes antes de começar o processo. Se você não tiver testes automatizados para a maior parte de sua aplicação, será necessário gastar algum tempo realizando testes manuais de todas as partes alteradas. No caso da atualização do Rails, isso significa cada umas das funcionalidades dentro da aplicação. Faça a si mesmo um favor e tenha certeza de ter uma boa cobertura de teste **antes** de iniciar uma atualização.
```
Sugiro trocar possuir para ter, assim a palavra possuir não aparece em duas frases consecutivas

### O Processo de Atualização

> Quando estiver atualizando a versão do Rails, o melhor é mover-se lentamente, uma versão *Minor* por vez, para fazer bom uso dos avisos de depreciação. As versões do Rails são numeradas da maneira *Major*.*Minor*.*Patch*. Versões *Major* e *Minor* tem permissão para alterar API pública, isso pode causar erros em sua aplicação. Versões *Patch* incluem apenas correções de *bug*, e não alteram nenhuma API pública.

Reviewer 2: ```suggestion
Quando estiver atualizando a versão do Rails, o melhor é ir devagar, uma versão *Minor* por vez, para fazer bom uso dos avisos de depreciação. As versões do Rails são numeradas da maneira *Major*.*Minor*.*Patch*. Versões *Major* e *Minor* têm permissão para alterar API pública, isso pode causar erros em sua aplicação. Versões *Patch* incluem apenas correções de *bug*, e não alteram nenhuma API pública.
```
Sugiro mudar a expressão "mover-se lentamente" para "ir devagar". Acredito que é mais do nosso cotidiano.
Também faltou um acento circunflexo no e do têm por serem dois ou mais elementos ;)

O processo deve correr da seguinte maneira:

1. Escreva os testes e garanta que eles passem.
2. Atualize para a ultima versão *Patch* após a versão atual de seu projeto.

> 3. Concerte os testes e funcionalidades  depreciadas.

Reviewer: Aqui parece que tem um espaço a mais, fiquei na dúvida

> 2. Atualize para a ultima versão *Patch* após a versão atual de seu projeto.

Reviewer 2: ```suggestion
2. Atualize para a última versão *Patch* após a versão atual de seu projeto.
```
Acho que ficou faltando o acento

> 3. Concerte os testes e funcionalidades depreciadas.

Reviewer 2: ```suggestion
3. Conserte os testes e funcionalidades depreciadas.
```

> 4. Atualize para a ultima versão *Patch* da versão *Minor* seguinte.

Reviewer 2: ```suggestion
4. Atualize para a última versão *Patch* da versão *Minor* seguinte.
```

> Repita este processo até chegar na versão desejada do Rails. Cada vez que você mudar entre versões. Você precisará alterar a versão no `Gemfile` (possivelmente também as versões de outras gems) e rodar `bundle update`. Rodar a tarefa de atualização mencionada abaixo para atualizar arquivos de configuração, e rode os testes.

Reviewer: ```suggestion
Repita este processo até chegar na versão desejada do Rails. Cada vez que você mudar entre versões, você precisará alterar a versão do Rails no `Gemfile` (e possivelmente também as versões de outras *gems*) e executar um `bundle update`. Após, execute a tarefa de atualização mencionada abaixo para atualizar arquivos de configuração, então execute os testes.
```

> Repita este processo até chegar na versão desejada do Rails. Cada vez que você mudar entre versões, Você precisará alterar a versão do Rails no `Gemfile` (possivelmente também as versões de outras *gems*) e executar um `bundle update`. Após, execute a tarefa de atualização mencionada abaixo para atualizar arquivos de configuração, então execute os testes.

Reviewer 2: ```suggestion
Repita este processo até chegar na versão desejada do Rails. Cada vez que você mudar entre versões, você precisará alterar a versão do Rails no `Gemfile` (possivelmente também as versões de outras *gems*) e executar um `bundle update`. Após, execute a tarefa de atualização mencionada abaixo para atualizar arquivos de configuração, então execute os testes.
```

Você pode encontrar a lista com todas as versões liberadas do Rails [aqui](https://rubygems.org/gems/rails/versions)

### Versões do Ruby

> Rails geralmente se mantém proximo à versão mais recente do ruby quando é liberado:

Reviewer: ```suggestion
Rails geralmente se mantém próximo à versão mais recente do ruby quando é liberado:
```

> Rails geralmente se mantém próximo à versão mais recente do ruby quando é liberado:

Reviewer 2: ```suggestion
Rails geralmente se mantém próximo à versão mais recente do Ruby quando é liberado:
```
Sugiro manter a Caixa Alta no nome de linguagens, como Ruby

* Rails 6 requer Ruby 2.5.0 ou posterior.
* Rails 5 requer Ruby 2.2.2 ou posterior.
* Rails 4 prefere Ruby 2.0 e requer 1.9.3 ou posterior.
* Rails 3.2.x é a última *branch* a suportar Ruby 1.8.7.
* Rails 3 e posteriores requerem Ruby 1.8.7 ou posteriores. O suporte para todas as versões anteriores do Ruby foi descontinuado. Você deve atualizar o quanto antes.

Dica: O Ruby 1.8.7 p248 e p249 possuem bugs que travam o Rails. Ruby Enterprise Edition possui correções para esses erros desde a versão 1.8.7-2010.02. Na Versão 1.9, Ruby 1.9.1 não é utilizável pois ela falha logo de início, então se você quiser utilizar a versão 1.9.x, pule diretamente para 1.9.3 para evitar problemas.

### A Tarefa de Atualização

> Rails fornece o comando `app:update` (`rake rails:update` na versão 4.2 e anteriores). Rode este comando após atualizar a versão do Rails no `Gemfile`. Isto lhe ajudará na criação de novos arquivos e na alteração de arquivos antigos em uma sessão interativa.

Reviewer: ```suggestion
Rails fornece o comando `app:update` (`rake rails:update` na versão 4.2 e anteriores). Execute este comando após atualizar a versão do Rails no `Gemfile`. Isto lhe ajudará na criação de novos arquivos e na alteração de arquivos antigos em uma sessão interativa.
```

1. Escreva os testes e garanta que eles passem.
2. Atualize para a ultima versão *Patch* após a versão atual de seu projeto.
3. Concerte os testes e funcionalidades  depreciadas.
4. Atualize para a ultima versão *Patch* da versão *Minor* seguinte.

Repita este processo até chegar na versão desejada do Rails. Cada vez que você mudar entre versões. Você precisará alterar a versão no `Gemfile` (possivelmente também as versões de outras gems) e rodar `bundle update`. Rodar a tarefa de atualização mencionada abaixo para atualizar arquivos de configuração, e rode os testes.

Você pode encontrar a lista com todas as versões liberadas do Rails [aqui](https://rubygems.org/gems/rails/versions)

### Versões do Ruby

Rails geralmente se mantém proximo à versão mais recente do ruby quando é liberado:

* Rails 6 requer Ruby 2.5.0 ou posterior.
* Rails 5 requer Ruby 2.2.2 ou posterior.
* Rails 4 prefers Ruby 2.0 and requires 1.9.3 or newer.
* Rails 3.2.x é a última *branch* a suportar Ruby 1.8.7.
* Rails 3 e posteriores requerem Ruby 1.8.7 ou posteriores. O suporte para todas as versões anteriores do Ruby foi descontinuado. Você deve atualizar o quanto antes.

> Dica: O Ruby 1.8.7 p248 e p249 possuem bugs que travam o Rails. Ruby Enterprise Edition possui correções para esses erros desde a versão 1.8.7-2010.02. Na Versão 1.9, Ruby 1.9.1 não é utilizável pois ela falha logo de início, então se você quiser utilizar a versão 1.9.x pule diretamente para 1.9.3 para evitar problemas.

Reviewer: Esse segfaults se não me engano era os segmentation faults da VM do YARV acho que podia ficar tipo: 'Devido aos problemas de memória' talvez. O @joaorsalmeida deve estar mais por dentro da época.

...
```

Não esqueça de revisar a diferença, para verificar se houveram mudanças inesperadas.

### Configurar Padrões de Framework

> A nova versão do rails pode ter configurações padrão diferentes da versão anterior. No entanto, após seguir os passos descritos acima, sua aplicação ainda estaria rodando com configurações padrão da versão **anterior** do Rails. Isso é porque o valor para `config.load_defaults` em `config/application.rb` ainda não foi alterado.

Reviewer 2: ```suggestion
A nova versão do Rails pode ter configurações padrão diferentes da versão anterior. No entanto, após seguir os passos descritos acima, sua aplicação ainda estaria rodando com configurações padrão da versão **anterior** do Rails. Isso porque o valor para `config.load_defaults` em `config/application.rb` ainda não foi alterado.
```
Sugiro colocar nome de linguagens e frameworks em Caixa Alta e baixa.
O "é" talvez não seja necessário por tirar a fluidez do texto.
